inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -0,0 +1,170 @@
+"""Protein solvation scenario."""
+from typing import Optional
+import os
+import shutil
+from inductiva.types import Path
+from inductiva.molecules.simulators import GROMACS
+from inductiva.utils.files import resolve_path, get_timestamped_path
+from inductiva.utils.misc import split_camel_case
+from inductiva.utils.templates import (TEMPLATES_PATH,
+                                       batch_replace_params_in_template)
+from inductiva.utils.files import remove_files_with_tag
+
+SCENARIO_TEMPLATE_DIR = os.path.join(TEMPLATES_PATH, "protein_solvation")
+GROMACS_TEMPLATE_INPUT_DIR = "gromacs"
+
+
+class ProteinSolvation():
+    """Solvated protein scenario."""
+
+    def __init__(self):
+        """Scenario constructor for protein solvation based on the 
+        GROMACS simulator.
+        The three main steps of this scenario are solvation, 
+        energy minimization and simulation. The user can control 
+        the number of steps used to perform the energy minimization step and 
+        the duration, temperature and integrator used to perform 
+        the simulation.
+        """
+        self.template_dir = os.path.join(SCENARIO_TEMPLATE_DIR,
+                                         GROMACS_TEMPLATE_INPUT_DIR)
+        self.simulator = GROMACS()
```

Comment:
I think the idea with the scenarios is that at some point they will be supported by multiple simulators, so I don't think simulator should be a property of the class but an argument of the simulate method.